src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractLanguageServerClient.cs

```diff
@@ -68,7 +68,7 @@ internal abstract class AbstractLanguageServerClient : ILanguageClient
 
         public Task<Connection> ActivateAsync(CancellationToken token)
         {
-            Contract.ThrowIfFalse(_languageServer == null, "This language server has already been initialized");
+            Contract.ThrowIfTrue(_languageServer?.Running == true, "The language server has not yet shutdown.");
```

Comment:
Is this really needed? Given the Activate() creates a new StreamJsonRpc and then the Shutdown/Exit are called _on that specific instance_, is there any reason we can't simply have two instances activated at the same time?

Reply:
We don't really ever want two instances of the same language server activated - the contract statement has caught bugs in the past around us / lsp client accidentally activating multiple times.

Reply:
But is this guarding our code (that we can't support this if it was ever needed) or is it just catching bugs in other components? Is there an API doc that says we can't expect multiple activations?

Reply:
Activation of the ILanguageClient triggers https://microsoft.github.io/language-server-protocol/specification#initialize which is spec'd to only allow initialize once per server.
Potentially for the inproclanguageserver we could support it, but for a connection to the OOP server, I don't know if we want to create a new connection every time (which reminds me, filed https://github.com/dotnet/roslyn/issues/44998 to support shutdown there)

Reply:
So "only once per server" is different than "there is only one server". If you squint and consider ActivateAsync to be "CreateANewServerAsync", is there any reason that's disallowed?

Reply:
So there is (was?) a contract that activateasync is called once per ILanguageClient instance (which is why the contract was originally written), though I couldn't find the document on it 
@tinaschrepfer that is still the case correct?